types/src/transaction/signature_verified_transaction.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    aggregator::AggregatorID,
+    contract_event::ContractEvent,
+    state_store::state_key::StateKey,
+    transaction::{BlockExecutableTransaction, Transaction},
+    write_set::WriteOp,
+};
+use move_core_types::language_storage::StructTag;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum SignatureVerifiedTransaction {
+    Valid(Transaction),
+    Invalid(Transaction),
+}
+
+impl SignatureVerifiedTransaction {
+    pub fn into_inner(self) -> Transaction {
+        match self {
+            SignatureVerifiedTransaction::Valid(txn) => txn,
+            SignatureVerifiedTransaction::Invalid(txn) => txn,
+        }
+    }
+
+    pub fn inner(&self) -> &Transaction {
+        match self {
+            SignatureVerifiedTransaction::Valid(txn) => txn,
+            SignatureVerifiedTransaction::Invalid(txn) => txn,
+        }
+    }
```

Comment:
make it this so that paranoid me feels safer :trollface: 
```
    pub fn expect_valid(&self) -> &Transaction {
        match self {
            SignatureVerifiedTransaction::Valid(txn) => txn,
            _ => panic!(),
        }
    }
```

Reply:
Done!

Reply:
Possible to remove `inner()` to make me feel safeR :trollface:? How do we justify getting the inner txn indifferently not caring if it's signature checked?

Reply:
This is mostly being used for some tests.. let me see if it can be doable..

Reply:
Removed the inner function.

Reply:
Please note that we still need the `into_inner` API as we store all txns in the chunk output. This is the same behavior we had previously as well.